examples/waldo/core/src/merkle.rs

```diff
@@ -21,12 +21,16 @@ use merkle_light::{
 };
 #[cfg(target_os = "zkvm")]
 use risc0_zkvm::guest;
-use risc0_zkvm::sha::{Digest, Impl, Sha256};
+use risc0_zkvm::{
+    declare_syscall,
+    sha::{Digest, Impl, Sha256},
+};
 use serde::{Deserialize, Serialize};
 
-/// RISC0 channel identifier for providing oracle access to a vector to the
-/// guest from the host.
-pub const VECTOR_ORACLE_CHANNEL: u32 = 0x09ac1e00;
+declare_syscall!(
+    /// RISC0 syscall for providing oracle access to a vector to the
+    /// guest from the host.
+    pub SYS_VECTOR_ORACLE);
```

Comment:
It feels like we're exposing a deeper interface to end-users. Channels feel like, these are standard ways to extend notion of I/O, and we're hiding the details of what it takes to do a syscall. I think it should be possible for advanced users to add their own syscalls, but I feel like channels are a nice intermediate level extension.

Reply:
I definitely like not asking the developer to pick their favorite number